### ISSUE TITLE
Efficient mouseless functionality

### ISSUE THREAD

**Issue by keyserj** [keyserj/ameliorate#5]
**Is your feature request related to a problem? Please describe.**
I want to be able to exercise ameliorate's functionality without a mouse - both for accessibility, and for efficiency. Specifically, at least the following should be mouseless:
* Add node
* Edit node text
* Edit selection's score
* View score's claim
* Go back to problem mapping from claim view
* Select node up/down/left/right of current selection
* Select edge up/down/left/right of current selection

**Describe the solution you'd like**
VIM-mode sounds pretty awesome (e.g. "cc" to change node text, "cs" change score, "apj" add problem below, "<esc>" to enter VIM-mode). Perhaps more common hotkeys like ctrl-[char] would be good for non-vimmers too.

**Describe alternatives you've considered**

**Additional context**

**Technical ideas**
No idea how this is normally done, but there must be a common way to implement shortcuts in React components. Theoretically, should be able to have `onKeyPress` or something, that would invoke the same method as an `onClick`. Unfortunately, a (very) brief search doesn't reveal any libraries for managing a vim mode for react components.

Manipulating edges may be a little more work than nodes because nodes already have a `selected` attribute. Can split that into a separate issue if that's desirable.

**Comment by keyserj** [comment]
Some more thoughts:
- j/k to navigate to nearest parent/child, h/l to navigate to nearest left/right node
	- how to navigate to edges? probably gj/gk
	- arrow keys easier to learn for newbies, but maybe worse when things like direction are needed elsewhere?
		- g[down]/g[up] seem not ideal
		- a[down]node-char/a[up]node-char seem not ideal
		- also they don't fit in the "[c]har that makes sense for char/vim-like command"
	- this should also work for navigating table cells
- enter to focus selected node's input
	- if focused, enter to blur, shift + enter to add newline
- shortcut to create parent/child
	- not sure what's best... possibly:
		- cp[node-type-char]/cc[node-type-char] for create parent/child
			- requires tracking a vim-like stack that can be escaped (and esc to deselect would have to be not possible unless nothing in vim-stack)
				- probably would want to add visuals to indicate what can be pressed, when the current stack is, option to turn off
		- cj[node-type-char]/ck[node-type-char] for create [direction]
		- aj[node-type-char]/ak[node-type-char] for add [direction]
			- a is closer to OG vim "append", whereas c is used for "change"
		- alt+p+[node-type-char]/alt+c+[node-type-char] for create parent/child?
			- hard to do triple shortcut actions that need to be held at the same time
			- but is nice that doesn't conflict with vimium i.e. require insert mode for vimium
	- what about when selecting a details/table node, not a diagram node?
		- maybe the add button itself could put "[p]arent"/"[c]hild" + "[n]ode-type-char" above/below itself
			- above if odd in container, below if even? to avoid text overlap
			- or could be _within_ the button too, temporarily replacing the icon
